pallets/passkey/src/lib.rs

```diff
@@ -106,6 +113,14 @@ pub mod module {
 			let main_origin = T::RuntimeOrigin::from(frame_system::RawOrigin::Signed(
 				transaction_account_id.clone(),
 			));
+
+			// check the signature
+			Self::check_account_signature(
```

Comment:
This should be in called in validate. I think having this check in here is duplicating the same check.

Reply:
Yes! I forgot about that, will move it in validate

Reply:
Done